src/compiler/checker.ts

```diff
@@ -9544,7 +9604,7 @@ namespace ts {
                     return getUnionType([leftType, rightType]);
                 case SyntaxKind.EqualsToken:
                     checkAssignmentOperator(rightType);
-                    return rightType;
+                    return getRegularTypeOfObjectLiteral(rightType);

```

Comment:
I added this because of code similar to the following in tsserver:

```typescript
interface A { a: number }
interface B extends A { b: number }
var last: B;
function foo(): A {
    return last = { a: 1, b: 2 };
}
```

Once the object literal is successfully assigned to a variable it seems pedantic to insist that `b` is an unknown property. And, really, following an assignment, the object literal really isn't "fresh" anymore.